crates/astria-sequencer-relayer/src/relayer.rs

```diff
@@ -56,13 +60,13 @@ pub struct Relayer {
     // A watch channel to track the state of the relayer. Used by the API service.
     state_tx: watch::Sender<State>,
 
-    // Sequencer blocks that have been received but not yet submitted to the data availability
-    // layer (for example, because a submit RPC was currently in flight) .
-    queued_blocks: Vec<SequencerBlockData>,
+    // Sequencer blocks that have been received but not yet finalized. Only finalized blocks are
+    // submitted to the data availability layer.
+    finalization_pipeline: FinalizationPipeline,
```

Comment:
the previous comment still stands right?
>     // (for example, because a submit RPC was currently in flight)

where do blocks that are finalized, but waiting for the current `submit_blob` RPC to finish, go? are they still in the `finalization_pipeline`?

Reply:
the previous comment stands, will add it back, thanks!
finalised blocks stay in the pipeline until it's drained
https://github.com/emhane/astria/blob/4abcbadf354e0b4d4352a027abc00570d70c3d85/crates/astria-sequencer-relayer/src/relayer.rs#L450